src/Altinn.App.Api/Controllers/DataController.cs

```diff
@@ -133,7 +134,7 @@ IPatchService patchService
             Application application = await _appMetadata.GetApplicationMetadata();
 
             DataType? dataTypeFromMetadata = application.DataTypes.First(e =>
-                e.Id.Equals(dataType, StringComparison.InvariantCultureIgnoreCase)
+                e.Id.Equals(dataType, StringComparison.Ordinal)
```

Comment:
Like this one, we were using `InvariantCultureIgnoreCase`, but some other places where the `Id` is compared we are using defaults